src/command.cc

```diff
@@ -125,6 +127,18 @@ void AutoStarVar::Eval(Evaluator*, string* s) const {
   pat.Stem(n->output.str()).AppendToString(s);
 }
 
+void AutoQuestionVar::Eval(Evaluator*, string* s) const {
+  unordered_set<StringPiece> seen;
+  WordWriter ww(s);
+  double target_age = GetTimestamp(ce_->current_dep_node()->output.str());
+  for (Symbol ai : ce_->current_dep_node()->actual_inputs) {
+    if (seen.insert(ai.str()).second
+      && GetTimestamp(ai.str()) > target_age) {
```

Comment:
For ninja mode, cannot we unconditionally output `ai`, ignoring timestamp? I think this feature is used to handle only updated inputs to save time (e.g., updating `.a` files). Ignoring timestamp may cause unnecessary work, but I guess it'd output the right thing for most cases? 

If we do this, we probably want `warn/werror` flags for people who do not want make-incompatible features?

Reply:
Yep, this sounds like the right approach, at least looks much simpler than the `ckati-newer` approach in the latest version.

Also, instead of `ckati-newer`, WDYT about doing this in bash snippet in the Ninja file, if you are okay with dealing with the file name quoting nightmare that would mean. That at least would mean one fewer moving part, which is very welcome.

Reply:
I agree that having $? work the same as $* would work in practice but I think it would be better to implement it correctly if we can do that in an acceptable way, especially since the goal of kati is to speed up builds.  Having slept on it for a while I think I can implement it in a much less drastic way, and like you said, without a helper program.